src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.cc

```diff
@@ -616,15 +607,18 @@ GrpcLb::PickResult GrpcLb::Picker::Pick(PickArgs args) {
   // If pick succeeded, add LB token to initial metadata.
   if (result.type == PickResult::PICK_COMPLETE &&
       result.subchannel != nullptr) {
+    const TokenAndClientStatsAttribute* attribute =
+        static_cast<const TokenAndClientStatsAttribute*>(
+            result.subchannel->GetAttribute(kGrpcLbAddressAttributeKey));
+    if (attribute == nullptr) {
+      gpr_log(GPR_ERROR, "[grpclb %p picker %p] No LB token for subchannel %p",
```

Comment:
minor nit: should this log message instead read something like:

"programming error, no TokenAndClientStatsAttribute for subchannel %p"

?

Just thinking:
a) this case reflects a missing LB token and also client stats configuration
b) except for a bug this should be impossible to trigger

Reply:
You're right that this could only happen as the result of a bug; it's essentially an assert, but with a slightly better error message.  This check is actually not new with this PR; all I'm doing here is changing it to check for the attribute instead of for a channel arg.  I kept the existing wording just because it doesn't seem worth changing; if we ever did get a bug report that indicated this message, we'd know where to look.